autogen/retrieve_utils.py

```diff
@@ -232,20 +232,67 @@ def is_url(string: str):
         return all([result.scheme, result.netloc])
     except ValueError:
         return False
-
-
-def create_vector_db_from_dir(
+        
+# New functions for LanceDB (similar functionality as ChromaDB functions)
+def create_lancedb_from_dir(
     dir_path: str,
     max_tokens: int = 4000,
-    client: API = None,
-    db_path: str = "/tmp/chromadb.db",
-    collection_name: str = "all-my-documents",
-    get_or_create: bool = False,
+    db_path: str = "/tmp/lancedb",
+    table_name: str = "pdf_search",
     chunk_mode: str = "multi_lines",
     must_break_at_empty_line: bool = True,
-    embedding_model: str = "all-MiniLM-L6-v2",
+    embedding_model_name: str = "all-MiniLM-L6-v2",
 ):
-    """Create a vector db from all the files in a given directory."""
+    db = LanceDB.connect(db_path)
+    try:
+        # Load embedding model
+        embeddings = HuggingFaceEmbeddings(model_name=embedding_model_name, model_kwargs={'device': 'cpu'})
```

Comment:
This `embeddings` is not used.

Reply:
should i use same emmeding which was used in chromadb?

Reply:
> should i use same emmeding which was used in chromadb?

Both are OK for me. It would be even better if you add support to customized embedding function. But that needs to be done with a parameter for users to pass in the embedding function.

The issue for your original code is that you defined the `embeddings`, but it's not used anywhere, because `embeddings` is assigned again in the following code.

Reply:
yes i removed this emnedding& using default one